src/popup/components/InviteItem.vue

```diff
@@ -82,85 +98,120 @@ export default {
           .href.replace(/^#/, ''),
         APP_LINK_WEB,
       );
-    },
-    address() {
-      return Crypto.getAddressFromPriv(this.secretKey);
-    },
-  },
-  watch: {
-    secretKey: {
-      async handler() {
-        await this.updateBalance();
-      },
-      immediate: true,
-    },
-  },
-  methods: {
-    formatDate,
-    deleteItem() {
-      this.$store.commit('invites/delete', this.secretKey);
-    },
-    async updateBalance() {
-      await watchUntilTruthy(() => this.sdk);
-      this.inviteLinkBalance = parseFloat(
-        await this.sdk
-          .balance(this.address, { format: AmountFormatter.AE_AMOUNT_FORMATS.AE })
-          .catch(() => 0),
+    });
+
+    const address = computed(() => Crypto.getAddressFromPriv(props.secretKey));
+
+    function deleteItem() {
+      root.$store.commit('invites/delete', props.secretKey);
+    }
+
+    async function updateBalance() {
+      const sdk = await getSdk();
+      inviteLinkBalance.value = parseFloat(
+        (await sdk
+          .balance(address.value, { format: AmountFormatter.AE_AMOUNT_FORMATS.AE })
+          .catch(() => 0)
+        )
+          .toString(),
       );
-    },
-    async claim() {
-      this.$emit('loading', true);
+    }
+
+    async function claim() {
+      emit('loading', true);
       try {
-        await this.$store.dispatch('invites/claim', this.secretKey);
-        await this.updateBalance();
+        await root.$store.dispatch('invites/claim', props.secretKey);
+        await updateBalance();
       } catch (error) {
-        if (await this.$store.dispatch('invites/handleNotEnoughFoundsError', { error, isInviteError: true })) return;
+        if (await root.$store.dispatch('invites/handleNotEnoughFoundsError', { error, isInviteError: true })) {
+          return;
+        }
         throw error;
       } finally {
-        this.$emit('loading', false);
+        emit('loading', false);
       }
-    },
-    resetTopUpChanges() {
-      this.topUpAmount = 0;
-      this.topUp = false;
-    },
-    async sendTopUp() {
-      this.$emit('loading', true);
+    }
+
+    /**
+     * Close the top up form and reset validator
+     * to avoid VeeValidate missing field errors.
+     */
+    async function resetTopUpChanges() {
+      formModel.value.amount = '';
+      $validator.pause();
+      await nextTick();
+      topUp.value = false;
+      $validator.reset();
+      $validator.resume();
+    }
+
+    async function sendTopUp() {
+      if (!(await $validator.validateAll())) {
+        return;
+      }
+      emit('loading', true);
+      const sdk = await getSdk();
       try {
-        if (this.topUpAmount > 0) {
-          await this.sdk.spend(this.topUpAmount, this.address, {
-            payload: 'referral',
-            denomination: AmountFormatter.AE_AMOUNT_FORMATS.AE,
-          });
-          await this.updateBalance();
+        await sdk.spend(formModel.value.amount, address.value, {
+          payload: 'referral',
```

Comment:
The `payload` is being set, and since the `formModel` doesn't have a `payload` set the `max` and `fee` is being calculated wrongly. The defined `payload` here is not doing anything except wasting money during this transfer. It is not defined in the sdk. It should be removed. Same in the `Invite` page.